### PR TITLE
Feature 4 adding grid unit test

### PR DESCRIPTION
**Pull request by orhanna14**
- Fix indentation to be from 4 -> 2
- Renames methods to make more sense
- Add spec for grid class

### CODE REVIEW

File: spec/grid_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    end
+
+    it "says an invalid coordinate is not in the coordinate values" do
+      coordinate = "Z6"
+      grid = Grid.new
+
+      expect(grid.coordinate_valid?(coordinate)).to eq false
+    end
+  end
+end
```
Comment (orhanna14) [inline spec/grid_spec.rb]: Is this too simple?  Though that might be the idea, test everything? It definitely is useful!

File: lib/game_runner.rb [hunk 2]
```diff
@@ -22,13 +22,13 @@ def run
   private
 
   def demand_valid_coordinates
-    while coordinates_invalid?
+    while !coordinate_valid?
```
Comment (baccigalupi) [inline lib/game_runner.rb]: This doesn't look like an improvement. I get that you want the symmetry between methods across classes, but more important is that you can read and understand a method responsibility quickly!

So, I would roll this back, but keep the `coordinate_valid?` method the same for the grid where it makes sense.

File: spec/grid_spec.rb (hunk 1 above)
Comment (baccigalupi) [reply spec/grid_spec.rb]: These are two scenarios, but if you categorize them they are:
1. Is valid in the grid somewhere
2. Is invalid if not on the grid

Whenever you have a boolean method like this, figuring out what each of the boundaries are and testing each of the boundaries is important for regression and also for documentation. These are possible boundaries, but you should figure out what is the most efficient way to test any possible boundary by testing only the ones the ones that are sufficient.

* The row is too low to be on the grid
* The row is too high to be on the grid
* The column is too low to be on the grid
* The column is to high to be on the grid
* The input is too many characters
* The input is too few characters
* The input doesn't have the right format
* What happens in the first character is lower case

You don't want to exhaustively test each coordinate on the grid, either. Just the boundaries.

**Review by baccigalupi — CHANGES_REQUESTED** [review]

File: lib/game_runner.rb (hunk 2 above)
Comment (orhanna14) [reply lib/game_runner.rb]: Got it!

File: spec/grid_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+require "spec_helper"
+require_relative "../lib/grid.rb"
+
+RSpec.describe Grid do
+  describe "#coordinate_valid?" do
```
Comment (baccigalupi) [inline spec/grid_spec.rb:5]: Each of these has an inside and an outside in each test, and I am not sure that in the right way to think about the boundary. Let's make those explicit in separate tests:

`it "is true if in the last row"`
`it "is false when after the last row"`

... which makes me think, is it true of false if it is before the first row?

**Review by baccigalupi — CHANGES_REQUESTED** [review]
Getting a lot better. Look at the comment on that first spec and use that to separate and flesh out the remaining specs.

**Pull request closed without merging**